### PR TITLE
mongo: Change mgo package to mongo-go-driver

### PR DESCRIPTION
**Pull request by Igoraamc**
What does this implement/fix? Explain your changes.
---------------------------------------------------
This pull request implements unit tests for all interactor features.

- Change some query entity types, to be compatible with the new driver
- Change the controller function that parses the string to int for int64
- Change the implementation of the data store to be compatible with the new driver

Does this close any currently open issues?
------------------------------------------
Any relevant logs, error output, etc?
-------------------------------------
Any other comments?
-------------------
Where has this been tested?
---------------------------

**Operating System/Platform:** MACOS Catalina

**GO Version:** go 1.13.7 darwin/amd64

### CODE REVIEW

File: pkg/data/data_store.go [hunk 1]
```diff
@@ -20,46 +22,48 @@ type Store interface {
...
 		return data, nil
 	}
 
-	err = ds.Database.C(collection).Find(findQuery).Select(bson.M{
-		"timestamp": 1,
-		"payload":   1,
-		"from":      1,
-	}).Skip(query.Skip).Sort(selectOrder).Limit(query.Take).All(&data)
+	options := ds.getFindOptions(query)
+
```
Comment (netoax) [inline pkg/data/data_store.go]: Please, remove this blank line.

File: pkg/data/data_store.go [hunk 2]
```diff
@@ -98,3 +102,19 @@ func (ds *store) getFindQuery(query *entities.Query) (bson.M, error) {
 
 	return b, nil
 }
+
+func (ds *store) getFindOptions(query *entities.Query) *options.FindOptions {
```
Comment (netoax) [inline pkg/data/data_store.go:105]: Very good! 👍

**Review by netoax — CHANGES_REQUESTED** [review]
Thanks for the PR, it seems ok for me. Please, just update the last line of the commit message in order to respect the 50/72 rule.

TIP: It's advisable to write dissertation-like messages, describing the main **motivation** and **what** changed (not how). Of course, you can if necessary use dashes or bullet points to explain topics or a list of things after describing the motivation.

File: pkg/data/data_store.go (hunk 1 above)
Comment (Igoraamc) [reply pkg/data/data_store.go]: Done!

File: pkg/data/data_store.go (hunk 2 above)
Comment (Igoraamc) [reply pkg/data/data_store.go:105]: Thank you!

**Review by lcbm — APPROVED** [review]
LGTM! 😄

**Review by netoax — APPROVED** [review]
LGTM!

**Pull request merged**